### DISCUSSION TITLE
TypeError: Cannot read property 'algolia' of undefined

### DISCUSSION THREAD

**Discussion by yokaiichi** [vuepress/core#286] [Q&A] [answered]
I took a risk and tried to upgrade a VuePress 1.x repo to Vuepress-next. (per Guide > Getting Started).
Essentially, I just ran `yarn add -D vuepress@next`. It completed sucessfully with only one warning:
```
warning "vuepress > @vuepress/theme-default > sass-loader@12.1.0" has unmet peer dependency "webpack@^5.0.0".
```
(Note that I never manually installed anything related to webpack back in my 1.x repo,)

So upon subsequently trying `yarn dev` for the first time, the dev server fails with the following errors:

```
yarn run v1.22.10
$ vuepress dev src
TypeError: Cannot read property 'algolia' of undefined
    at module.exports (C:\Users\sgrey\Documents\GitHub\greywalker-notebook\docs\node_modules\@vuepress\theme-default\index.js:9:17)
    at Object.resolvePlugin (C:\Users\sgrey\Documents\GitHub\greywalker-notebook\docs\node_modules\@vuepress\core\lib\app\resolvePlugin.js:18:11)
    at Object.resolveTheme (C:\Users\sgrey\Documents\GitHub\greywalker-notebook\docs\node_modules\@vuepress\core\lib\app\resolveTheme.js:19:35)
    at Object.resolveThemeApi (C:\Users\sgrey\Documents\GitHub\greywalker-notebook\docs\node_modules\@vuepress\core\lib\app\resolveThemeApi.js:11:34)
    at Object.createBaseApp (C:\Users\sgrey\Documents\GitHub\greywalker-notebook\docs\node_modules\@vuepress\core\lib\app\createBaseApp.js:43:40)
    at Object.createDevApp (C:\Users\sgrey\Documents\GitHub\greywalker-notebook\docs\node_modules\@vuepress\core\lib\app\createDevApp.js:10:33)
    at dev (C:\Users\sgrey\Documents\GitHub\greywalker-notebook\docs\node_modules\@vuepress\cli\lib\commands\dev\createDev.js:27:28)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Note that I never installed/implemented anything related to Algolia in v1.x

I tried googling this error and poking around in the listed files, but I'm not a Vue dev; I'm a tech writer, so debugging this is beyond me.

Am I seeing a bug in vuepress-next, or is this a fixable thing? Right now my site's dead because I can't spin up the dev server.

BTW this is on Windows 10.

Further clues might be in my **package.json**? After attempting to upgrade my repo with `yarn add -D vuepress@next`, it looks like the `devDependencies` object still lists only v 1.x versions of all the basic plugins?  Also, note that the `docsearch` plugin is not listed (but the Algolia error might be expecting that plugin?

```
  "devDependencies": {
    "@vuepress/plugin-active-header-links": "^1.8.2",
    "@vuepress/plugin-back-to-top": "^1.8.2",
    "@vuepress/plugin-medium-zoom": "^1.8.2",
    "@vuepress/plugin-nprogress": "^1.8.2",
    "@vuepress/plugin-register-components": "^1.8.2",
    "@vuepress/plugin-search": "^1.8.2",
    "@vuepress/theme-default": "^1.8.2",
    "vuepress": "^2.0.0-beta.22",
    "vuepress-plugin-container": "^2.1.5"
  }
```

The **Official Plugins Reference** section of the 2.x doc would imply that all of these should have `@next` appended to all of them. 
* Is it possible the **Getting Started** instructions are missing some step for bulk installing all the required package dependencies if you're trying to upgrade a 1.x repo to 2.x?
* Or is there a bug with `yarn add -D vuepress@next` that isn't correctly installing packages on Windows systems?
* Should I wait for a potential update or just go manually run `yarn add` for all the `*@next` versions of all the above-listed plugins?

**Comment by meteorlxy** [comment, marked as answer]
Your plugins and themes are for 1.x, which are not compatible with 2.x

**Reply by orefalo** [reply]
I have the same issue...

**Reply by orefalo** [reply]
Confusing instructions lead to this outcome.

**Reply by orefalo** [reply]
If I remove 
//    "@vuepress/plugin-back-to-top": "^1.8.2",
//    "@vuepress/plugin-google-analytics": "^1.8.2",
//    "@vuepress/plugin-medium-zoom": "^1.8.2",
//    "@vuepress/plugin-pwa": "^1.8.2",
//    "@vuepress/plugin-register-components": "^1.8.2",
//    "@vuepress/plugin-search": "^1.8.2",
from package.json.

then npm run dev complains about plugins not found

**Comment by ramesh-dada** [comment]
> If I remove
> // "@vuepress/plugin-back-to-top": "^1.8.2",
> // "@vuepress/plugin-google-analytics": "^1.8.2",
> // "@vuepress/plugin-medium-zoom": "^1.8.2",
> // "@vuepress/plugin-pwa": "^1.8.2",
> // "@vuepress/plugin-register-components": "^1.8.2",
> // "@vuepress/plugin-search": "^1.8.2",
> from package.json.
> then npm run dev complains about plugins not found
> 
No need to remove just update `npm update @vuepress/plugin-name@next` In this command just change the @vuepress/plugin-name to package name you want to update so that you will be able to update all of your plugins or you can simply type the required version in package.json file and then simply run `npm update`. If you are problem is solved select my post as an answer.
